### PR TITLE
Add a mini-hlint expect test

### PR DESCRIPTION
**Pull request by shayne-fletcher**
- Sync to `6db8a0f76ec45d47060e28bb303e9eef60bdb16b`
- Get a windows `ghc-master` ghc-8.10.4 build going  in CI 
- Add a `--no-checkout` argument to `CI.hs`
  - When it's passed:
    - Don't `rm -rf ghc` on startup
    - Don't `git clone` just refresh existing clone
- Various`mini-hlint` improvements
  - Fix various broken things
  - Write a golden test-suite for it
    - Run that rather than mini-hlint directly in CI

### CODE REVIEW

File: CI.hs [hunk 1]
```diff
@@ -217,26 +224,53 @@ buildDists
...
+            Ghc882 -> cmd "cd ghc && git fetch --tags && git checkout ghc-8.8.2-release"
+            Ghc883 -> cmd "cd ghc && git fetch --tags && git checkout ghc-8.8.3-release"
+            Ghc884 -> cmd "cd ghc && git fetch --tags && git checkout ghc-8.8.4-release"
+            Da { mergeBaseSha, patches, upstream } -> do
+                cmd $ "cd ghc && git fetch --tags && git checkout " <> mergeBaseSha
+                -- Apply Digital Asset extensions.
+                cmd $ "cd ghc && git remote add upstream " <> upstream
+                cmd "cd ghc && git fetch upstream"
+                cmd $ "cd ghc && git -c user.name=\"Cookie Monster\" -c user.email=cookie.monster@seasame-street.com merge --no-edit " <> unwords patches
+            GhcMaster hash -> cmd $ "cd ghc && git checkout " ++ hash
```
Comment (cocreature) [inline CI.hs:272]: This is duplicating 95% of the logic but the only difference is that in one case we call `git clone` (and for some reason `git fetch --tags`?) Can we share the 95% by calling the `git clone` step separately beforehand?

**Review by cocreature — APPROVED** [review]
Thanks!

File: CI.hs (hunk 1 above)
Comment (shayne-fletcher) [reply CI.hs:272]: absolutely. done.

Comment (shayne-fletcher) [reply CI.hs]: mind giving this an `/azp run` please?

Comment (shayne-fletcher) [reply CI.hs]: oops. introduced an error. sorry. fixed now.

**Pull request merged**